server/src/main/java/org/folio/metastorage/server/OaiPmhClient.java

```diff
@@ -0,0 +1,469 @@
+    if (userHeaders != null) {
+      userHeaders.forEach(e -> {
+        if (e.getValue() instanceof String value) {
+          headers.add(e.getKey(), value);
+        } else {
+          throw new IllegalArgumentException("headers " + e.getKey() + " value must be string");
+        }
+      });
+    }
+    return headers;
+  }
+
+  static boolean addQueryParameterFromConfig(QueryStringEncoder enc,
+      JsonObject config, String key) {
+    String value = config.getString(key);
+    if (value == null) {
+      return false;
+    }
+    enc.addParam(key, value);
+    return true;
+  }
+
+  static void addQueryParameterFromParams(QueryStringEncoder enc, JsonObject params) {
+    if (params != null) {
+      params.forEach(e -> {
+        if (e.getValue() instanceof String value) {
+          enc.addParam(e.getKey(), value);
+        } else {
+          throw new IllegalArgumentException("params " + e.getKey() + " value must be string");
+        }
+      });
+    }
+  }
+
+  Future<Void> ingestRecord(
+      Storage storage, OaiRecord<JsonObject> oaiRecord,
+      SourceId sourceId, JsonArray matchKeyConfigs) {
+    try {
+      JsonObject globalRecord = new JsonObject();
+      globalRecord.put("localId", oaiRecord.getIdentifier());
+      if (oaiRecord.isDeleted()) {
+        globalRecord.put("delete", true);
+      } else {
+        globalRecord.put("payload", new JsonObject().put("marc", oaiRecord.getMetadata()));
+      }
+      return storage.ingestGlobalRecord(vertx, sourceId, globalRecord, matchKeyConfigs);
+    } catch (Exception e) {
+      log.error("{}", e.getMessage(), e);
+      return Future.failedFuture(e);
+    }
+  }
+
+  static class Datestamp {
+    String value;
+  }
+
+  void oaiHarvestLoop(Storage storage, SqlConnection connection, String id, JsonObject job) {
+    JsonObject config = job.getJsonObject(CONFIG_LITERAL);
+    RequestOptions requestOptions = new RequestOptions();
+    requestOptions.setMethod(HttpMethod.GET);
+    requestOptions.setHeaders(getHttpHeaders(config));
+    QueryStringEncoder enc = new QueryStringEncoder(config.getString("url"));
+    enc.addParam("verb", "ListRecords");
+    if (!addQueryParameterFromConfig(enc, config, RESUMPTION_TOKEN_LITERAL)) {
+      addQueryParameterFromConfig(enc, config, "from");
+      addQueryParameterFromConfig(enc, config, "until");
+      addQueryParameterFromConfig(enc, config, "set");
+      addQueryParameterFromConfig(enc, config, "metadataPrefix");
+    }
+    addQueryParameterFromParams(enc, config.getJsonObject("params"));
+    String absoluteUri = enc.toString();
+    log.info("client send request {}", absoluteUri);
+    requestOptions.setAbsoluteURI(absoluteUri);
+
+    storage.getAvailableMatchConfigs(connection)
+        .compose(matchKeyConfigs ->
+            httpClient.request(requestOptions)
+                .compose(HttpClientRequest::send)
+                .compose(res -> {
+                  log.info("client handle response");
```

Comment:
Clean up or provide context to the message (e.g jobId or URL)

Reply:
Removed it.